--- CMakeLists.txt.orig	2011-04-14 01:17:22.000000000 +0900
+++ CMakeLists.txt	2011-04-25 19:13:10.000000000 +0900
@@ -94,6 +94,7 @@
 vtkNetCDFCFReader.cxx
 vtkNetCDFPOPReader.cxx
 vtkNetCDFReader.cxx
+vtkOFFDevReader.cxx
 vtkOBJReader.cxx
 ${_VTK_OGGTHEORA_SOURCES}
 vtkOpenFOAMReader.cxx
@@ -402,6 +403,14 @@
 ENDIF ( VTK_USE_ODBC AND ODBC_FOUND )
 
 #-----------------------------------------------------------------------------
+# Regex
+# See if a POSIX Regex library exists on this system. Used for
+# vtkNewOpenFOAMReader.
+#
+INCLUDE(${VTK_CMAKE_DIR}/FindRegex.cmake)
+
+
+#-----------------------------------------------------------------------------
 # Add these classes only if vtkTypeUInt64Array exists
 IF(VTK_HAS_UINT64_ARRAY)
   SET(Kit_SRCS
