--- CMakeLists.txt.orig	2010-05-28 02:07:13.000000000 +0900
+++ CMakeLists.txt	2010-06-06 21:20:48.880129308 +0900
@@ -143,6 +143,7 @@
   pqOrbitWidget.h
   pqOutputPortComboBox.h
 #  pqParticleTracerPanel.h
+  pqPOFFDevReaderPanel.h
   pqPQLookupTableManager.h
   pqPipelineBrowserWidget.h
   pqPipelineModel.h
@@ -553,6 +554,8 @@
   pqOrbitWidget.h
   pqOutputPortComboBox.cxx
   pqOutputPortComboBox.h
+  pqPOFFDevReaderPanel.cxx
+  pqPOFFDevReaderPanel.h
   pqPQLookupTableManager.cxx
   pqPQLookupTableManager.h
 #  pqParticleTracerPanel.cxx
